Engine/DataFeeds/TimeSliceFactory.cs

```diff
@@ -0,0 +1,524 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Collections.Generic;
+using NodaTime;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+
+namespace QuantConnect.Lean.Engine.DataFeeds
+{
+    /// <summary>
+    /// Instance base class that will provide methods for creating new <see cref="TimeSlice"/>
+    /// </summary>
+    public class TimeSliceFactory
+    {
+        /// <summary>
+        /// The time zone required for computing algorithm and slice time
+        /// </summary>
+        public DateTimeZone TimeZone { get; set; }
```

Comment:
Maybe the slice time zone could be passed to the constructor and the `TimeZone` property be `get` only?

Reply:
Your right, it will look cleaner, will do 👍